charts/karpenter-core/templates/clusterrole.yaml

```diff
@@ -50,7 +50,7 @@ rules:
   # Write
   - apiGroups: ["karpenter.sh"]
     resources: ["provisioners/status", "machines", "machines/status"]
-    verbs: ["create", "delete", "patch"]
+    verbs: ["create", "delete", "update", "patch"]
```

Comment:
this is the first file in the changeset, so apologies if this is obvious as I progress downwards, but what specifically about this new controller requires update in addition to patch for the karpenter ClusterRole?

Reply:
The main thing is that we haven't done optimistic locking with update up to this point so we haven't had a need for the permission. They are effectively the same thing (in a way) since they both allow you to modify a resource. I actually have no idea behind the history of why these are treated as different permissions in the ClusterRole.